python-checks/src/main/java/org/sonar/python/checks/TorchModuleShouldCallInitCheck.java

```diff
@@ -0,0 +1,106 @@
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.FunctionDef;
+import org.sonar.plugins.python.api.tree.QualifiedExpression;
+import org.sonar.plugins.python.api.tree.RegularArgument;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.python.checks.utils.CheckUtils;
+import org.sonar.python.quickfix.TextEditUtils;
+import org.sonar.python.tree.TreeUtils;
+
+@Rule(key = "S6978")
+public class TorchModuleShouldCallInitCheck extends PythonSubscriptionCheck {
+  private static final String TORCH_NN_MODULE = "torch.nn.Module";
+  private static final String MESSAGE = "Add a call to super().__init__()";
+  private static final String SECONDARY_MESSAGE = "Inheritance happens here";
+  public static final String QUICK_FIX_MESSAGE = "insert call to super constructor";
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.FUNCDEF, ctx -> {
+      FunctionDef funcDef = (FunctionDef) ctx.syntaxNode();
+      ClassDef classDef = CheckUtils.getParentClassDef(funcDef);
+      if (isInheritingFromTorchModule(classDef) && isConstructor(funcDef) && isMissingSuperCall(funcDef)) {
+        PreciseIssue issue = ctx.addIssue(funcDef.name(), MESSAGE);
+        issue.secondary(classDef.name(), SECONDARY_MESSAGE);
+        createQuickFix(funcDef).ifPresent(issue::addQuickFix);
+      }
+    });
+  }
+
+  private static boolean isConstructor(FunctionDef funcDef) {
+    FunctionSymbol symbol = TreeUtils.getFunctionSymbolFromDef(funcDef);
+    return symbol != null && "__init__".equals(symbol.name()) && funcDef.isMethodDefinition();
+  }
+
+  private static boolean isInheritingFromTorchModule(@Nullable ClassDef classDef) {
+    if (classDef == null) return false;
+    ArgList args = classDef.args();
+    return args != null && args.arguments().stream()
+      .flatMap(TreeUtils.toStreamInstanceOfMapper(RegularArgument.class))
+      .map(arg -> getQualifiedName(arg.expression()))
+      .anyMatch(expr -> expr.filter(TORCH_NN_MODULE::equals).isPresent());
+  }
+
+  private static Optional<String> getQualifiedName(Expression node) {
+    return TreeUtils.getSymbolFromTree(node).flatMap(symbol -> Optional.ofNullable(symbol.fullyQualifiedName()));
+  }
+
+  private static boolean isMissingSuperCall(FunctionDef funcDef) {
+    return !TreeUtils.hasDescendant(funcDef, t -> t.is(Tree.Kind.CALL_EXPR) && isSuperConstructorCall(((CallExpression) t)));
+  }
+
+  private static boolean isSuperConstructorCall(CallExpression callExpr) {
+    return callExpr.callee() instanceof QualifiedExpression qualifiedCallee && isSuperCall(qualifiedCallee.qualifier()) && "__init__".equals(qualifiedCallee.name().name());
+  }
+
+  private static boolean isSuperCall(Expression qualifier) {
+    if (qualifier instanceof CallExpression callExpression) {
+      Symbol superSymbol = callExpression.calleeSymbol();
+      return superSymbol != null && "super".equals(superSymbol.name());
```

Comment:
Here if you import a function called super or pass it as a parameter of the `__init__` method wouldn't you get a null symbol for the super call expression?